tests/integration/tests/conftest.py

```diff
@@ -44,10 +43,32 @@ def fixture_data(start_stop_cloudbeat, k8s, cloudbeat_agent):
     @param start_stop_cloudbeat: fixture to start and stop agent / cloudbeat
     @param k8s: Kubernetes wrapper object
     @param cloudbeat_agent: config object
-    @return: pods, nodes in cluster
+    @return: pods, nodes, leader_node in cluster
     """
     # pylint: disable=W0612,W0613
     pods = k8s.get_agent_pod_instances(agent_name=cloudbeat_agent.name,
                                        namespace=cloudbeat_agent.namespace)
     nodes = k8s.get_cluster_nodes()
-    return pods, nodes
+    leader_node = get_cluster_leader(k8s, cloudbeat_agent, pods)
+
+    return pods, nodes, leader_node
+
+
+def get_cluster_leader(k8s, cloudbeat_agent, pods):
+    lease_name = "cloudbeat-cluster-leader"
```

Comment:
These tests shall be relevant for cloudbeat and elastic agent. If lease_name will not be the same please find better solution.

Reply:
It will be the same for both.